src/python/linear regression.py

```diff
@@ -0,0 +1,336 @@
+
+# In[31]:
+
+
+ax1 = hm.plot.scatter(x='H3K4me1',
+                      y='Cell Line',
+                      c='DarkBlue')
+# ax1.figure.savefig('H3K4me1.pdf')
+ax2 = hm.plot.scatter(x='H3K9ac1K14ac0',
+                      y='Cell Line',
+                      c='Red')
+
+
+# ### Eliminate NaN
+# ### Split GCP and MET into train and test set respectively.
+
+# In[32]:
+
+
+GCP = np.nan_to_num(GCP, nan=0)
+MET = np.nan_to_num(MET, nan=0)
+Xtrain, Xtest, Ytrain, Ytest =train_test_split(GCP, MET, test_size=0.3, random_state=0)
+
+print(Xtest.shape)
+
+print(Ytrain.shape)
+
+
+# ### Fit train sets with Linear regression to generate models
+
+# In[4]:
+
+
+GCP2MET_models = []
+for i in range(Ytrain.shape[1]):
+    mdl_G2M = LinearRegression().fit(Xtrain, Ytrain[:, i])
+    GCP2MET_models.append(mdl_G2M)
+
+
+MET2GCP_models = []
+
+for j in range(Xtrain.shape[1]):
+    mdl_M2G = LinearRegression().fit(Ytrain, Xtrain[:,j])
+    MET2GCP_models.append(mdl_M2G)
+
+
+# In[5]:
+
+
+# def robust_zscore(df):
+#     """
+#     robust_zscore computes a modified version of the Z-score centered around the media and scaled by the median absolute deviation.
+#     INPUT:
+#       :param df: A data frame corresponding to numerical values where rows correspond to cells and columns correspond to genes.
+#     OUTPUTS:
+#       :output Z: A data frame of the robust Z-scores.
+#       :output pval: A data frame of the p-values corresponding to the robust Z-scores.
+#     # Compute model parameters
+#     """
+#     med = df.median(axis=0)
+#     n = df.shape[1]
+#     med_abs_dev = mad(df)
+#     scale_factor = 1.4826
+#     # Compute robust Z-score
+#     Z = (df - med) / (med_abs_dev * scale_factor)
+#     Z[~np.isfinite(Z)] = 0
+#     # Compute P-value based on Z-score
+#     pval = norm.sf(abs(Z))*2 
+#     pval = pd.DataFrame(pval,
+#                     index=df.index,
+#                     columns=df.columns)
+#     return(Z, pval)
+
+
+# ### To evaluate models and get correlation coefficient, pvalue, and mse value.
+
+# In[42]:
+
+
+def evaluate_models(models, Xtest, Ytest):
+    """
+    evaluate_models returns results from the model predictions, including the pearson
+    correlation coefficient, p-Values, and MSE.
+
+    :param models:         A list of scikit-learn model objects.
+    :param Xtest:          A numpy array or pandas dataframe containing validation set input data.
+    :param Ytest:          A numpy array or pandas dataframe containing validation set output data.
+    :return pred_resul:    A dictionary containing the final MSE, pValue, or rValue.
+    
+    """
+    
+    predictions = []
+    rValue = list()
+    pValue = list()
+    MSE = list()
+    for i in range(len(models)):
+        mdl = models[i]
+        Ypred = mdl.predict(Xtest)
+        predictions.append(Ypred)   
+    
+        r, p_value = pearsonr(Ypred, Ytest[:, i])
+        # Compute P-value based on r-value
+        pval = norm.sf(abs(r))*2 
```

Comment:
Ah, this may be where things are going wrong. I would run this code once you get a vector of r values out of the loop. I think this will return different values because your distribution will be different. I may be wrong, but it's worth a shot.